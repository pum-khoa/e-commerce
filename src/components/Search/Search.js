import { CloseOutlined, SearchOutlined } from "@ant-design/icons";
import { useEffect } from "react";
import "./Search.scss";

export default function Search(props) {
  const { searchInput, searchStatus, hideSearch, onSearch, onChangeInput } =
    props;

  useEffect(() => {
    const searchWrapper = document.querySelector(".search__wrapper");
    const search = document.querySelector(".search");
    if (searchStatus) {
      searchWrapper.classList.add("search__wrapper--display");
      search.classList.add("search--display");
    } else {
      setTimeout(() => {
        searchWrapper.classList.remove("search__wrapper--display");
      }, 200);
      search.classList.remove("search--display");
    }
  }, [searchStatus]);

  return (
    <div className="search__wrapper">
      <div className="search">
        <div className="search__search-btn" onClick={onSearch}>
          <SearchOutlined />
        </div>
        <input
          value={searchInput}
          type="text"
          className="search__input"
          placeholder="Search our store"
          onInput={(e) => {
            onChangeInput(e);
          }}
        />
        <div className="search__close-btn" onClick={hideSearch}>
          <CloseOutlined />
        </div>
      </div>
      <div className="search__overlays" onClick={hideSearch}></div>
    </div>
  );
}
