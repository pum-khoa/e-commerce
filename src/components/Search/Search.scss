.search__wrapper {
    display: flex;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: -1;
    opacity: 0;
    background: rgba(0, 0, 0, 0.6);
    flex-direction: column;
}

.search__wrapper--display {
    z-index: 1;
    opacity: 1;
}

.search {
    height: 120px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 40px 20px;
    background: #fff;
    transform: translateY(-100%);
    transition: all 0.2s ease-out;
    opacity: 0;
}

.search__overlays {
    flex: 1;
}

.search--display {
    transform: translateY(0);
    opacity: 1;
}

.search__search-btn,
.search__close-btn {
    padding: 1px 6px;
    font-size: 1.5rem;
    cursor: pointer;
}

.search__input {
    flex: 1;
    padding: 8px 16px;
    outline: none;
}
