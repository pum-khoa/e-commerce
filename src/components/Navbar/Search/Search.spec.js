import ReactTestRenderer from 'react-test-renderer';

import Search from './Search';

describe('Navbar/Search', () => {
  it('Should render correctly', () => {
    const renderer = ReactTestRenderer.create(<Search />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
