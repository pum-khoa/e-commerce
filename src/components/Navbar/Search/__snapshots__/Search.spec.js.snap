// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar/Search Should render correctly 1`] = `
<div
  className="search__wrapper"
>
  <div
    className="search"
  >
    <div
      className="search__search-btn"
    >
      <span
        aria-label="search"
        className="anticon anticon-search"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </span>
    </div>
    <input
      className="search__input"
      onInput={[Function]}
      placeholder="field.search_nav.placeholder"
      type="text"
    />
    <div
      className="search__close-btn"
    >
      <span
        aria-label="close"
        className="anticon anticon-close"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="close"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
          />
        </svg>
      </span>
    </div>
    <div
      className="searchList__container"
    >
      <ul
        className="searchList__list"
      />
    </div>
  </div>
  <div
    className="search__overlays"
  />
</div>
`;
